gax/src/test/java/com/google/api/gax/rpc/RetryingTest.java

```diff
@@ -82,7 +83,7 @@ public void resetClock() {
     clientContext =
         ClientContext.newBuilder()
             .setExecutor(executor)
-            .setClock(fakeClock)
+            .setClock(NanoClock.getDefaultClock())
```

Comment:
Why is fake clock replaced with real clock? When real clock is used in tests it usually easily leads to flaky tests (different machines and same machine under different circumstances process things with different speed, and if tests depend on real time the whole thing goes out of control).

Reply:
I had added a real clock to test out the scenario with only `maxAttempts` is set. With fake clock was retrying forever because [`totalTimeSpentNanos`](https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java#L133-L141) ended up 0 everytime.

Reply:
It is a fake clock, so you can modify it as you need. Can we still use fake clock and achieve the result you want (i.e. test scenario when only maxAttempts is set).

Reply:
Thanks for the suggestion, I have changed the test case to use `FakeApiClock` now.